src/WorkflowCore/Services/ExecutionResultProcessor.cs

```diff
@@ -41,6 +41,7 @@ public void ProcessExecutionResult(WorkflowInstance workflow, WorkflowDefinition
             {
                 pointer.EventName = result.EventName;
                 pointer.EventKey = result.EventKey;
+                pointer.EventPublished = false;
```

Comment:
I believe this would cause existing workflows to behave unexpectedly - we should probably write some tests to guard against that... under normal circumstances, a `WaitFor` should only fire once.

You could implement your own alternate `ReWaitFor` primitive... have a look at https://github.com/danielgerlag/workflow-core/blob/master/src/WorkflowCore/Primitives/WaitFor.cs

When it returns `ExecutionResult.Next()`, it will no longer wait for any further events, but when it returns `ExecutionResult.WaitForEvent(EventName, EventKey, effectiveDate)`, it will wait for more... you can adapt this to your needs.  You can also look at the fluent builder extension method to see how to call it - https://github.com/danielgerlag/workflow-core/blob/master/src/WorkflowCore/Services/FluentBuilders/StepBuilder.cs#L133

Let me know if you need more info.

Reply:
I don't think only doing the primitive will work due to line 89 in EventConsumer, if the event published flag doesn't get reset, the subscription doesn't get built for the next event.  Also, I don't think this method will make sense in the fluent builder since there are whiles to make this scenario work.  I think it's only a response to a custom step, that I need to wait for the event again.  Does that make sense?

I will commit a different change that will isolate the logic to only be for a call to ExecutionResult.ReWaitFor(...).

Let me know your thoughts.